src/pages/track_chairs/team-lists-page.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright 2017 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * */
+
+import React, { useEffect, useMemo } from "react";
+import { connect } from "react-redux";
+import T from "i18n-react/dist/i18n-react";
+import { Breadcrumb } from "react-breadcrumbs";
+import { FreeTextSearch } from "openstack-uicore-foundation/lib/components";
+import {
+  getSelectionPlans,
+  getSourceList,
+  getTeamList,
+  reorderList,
+  setSelectionPlan,
+  updateTeamList
+} from "../../actions/track-chair-actions";
+import TrackDropdown from "../../components/inputs/track-dropdown";
+import SelectionPlanDropdown from "../../components/inputs/selection-plan-dropdown";
+import List from "../../components/dnd-list/List";
+import { moveItem } from "../../utils/methods";
+import styles from "../../styles/team-list-page.module.less";
+
+const getTracksFromSelectionPlan = (selectionPlan) => 
+  // get all tracks from the selection plan and track visible
+   selectionPlan?.track_groups
+    .reduce((result, trackGroup) => {
+      trackGroup?.tracks?.forEach((track) => {
+        if (!result.find((tr) => tr.id === track.id)) {
+          result.push(track);
+        }
+      });
+      return result;
+    }, [])
+    .filter((t) => t?.chair_visible)
+;
+
+const TeamListsPage = ({
+  summit,
+  match,
+  selectionPlans,
+  sourceList,
+  teamList,
+  sourceSearchTerm,
+  sourceTrackId,
+  sourceSelPlanId,
+  teamSelPlanId,
+  teamTrackId,
+  sourcePage,
+  sourceLastPage,
+  ...props
+}) => {
+  const sourceTrackOptions =
+    useMemo(
+      () =>
+        getTracksFromSelectionPlan(
+          selectionPlans.find((sp) => sp.id === sourceSelPlanId)
+        ),
+      [sourceSelPlanId]
+    ) || [];
+  const teamTrackOptions =
+    useMemo(
+      () =>
+        getTracksFromSelectionPlan(
+          selectionPlans.find((sp) => sp.id === teamSelPlanId)
+        ),
+      [teamSelPlanId]
+    ) || [];
+
+  useEffect(() => {
+    if (!selectionPlans.length) {
+      props.getSelectionPlans();
```

Comment:
this is querying the entire data set check https://github.com/fntechgit/summit-admin/pull/587/files#r2078090446